generate-offline-map/src/main/java/com/esri/arcgismaps/sample/generateofflinemap/components/MapViewModel.kt

```diff
@@ -0,0 +1,273 @@
+                    graphicsOverlay.graphics.add(downloadArea)
+
+                    // Limit the map scale to the largest layer scale
+                    map.maxScale = map.operationalLayers[6].maxScale ?: 0.0
+                    map.minScale = map.operationalLayers[6].minScale ?: 0.0
+                }
+                .onFailure {
+                    messageDialogVM.showMessageDialog(
+                        title = it.message.toString()
+                    )
+                }
+        }
+    }
+
+    /**
+     * Generate an offline map job with the default [GenerateOfflineMapParameters]
+     *
+     */
+    fun createOfflineMapJob() {
+        // Offline map path
+        val offlineMapPath =
+            application.getExternalFilesDir(null)?.path + File.separator + "offlineMap"
+
+        // Delete any offline map already in the cache
+        File(offlineMapPath).deleteRecursively()
+
+        // Specify the extent, min scale, and max scale as parameters
+        var minScale: Double = map.minScale ?: 0.0
+        val maxScale: Double = map.maxScale ?: 0.0
+
+        // Variable minScale must always be larger than maxScale
+        if (minScale <= maxScale) {
+            minScale = maxScale + 1
+        }
+
+        // Get the geometry of the downloadArea
+        val geometry = downloadArea.geometry ?: return messageDialogVM.showMessageDialog(
+            title = "Could not get geometry of the downloadArea"
+        )
+
+        // Set the offline map parameters
+        val generateOfflineMapParameters = GenerateOfflineMapParameters(
+            areaOfInterest = geometry,
+            minScale = minScale,
+            maxScale = maxScale
+        ).apply {
+            // Set job to cancel on any errors
+            continueOnErrors = false
+        }
+
+        // Create an offline map task with the map
+        val offlineMapTask = OfflineMapTask(onlineMap = map)
+
+        viewModelScope.launch {
+            offlineMapTask.load().getOrElse {
+                messageDialogVM.showMessageDialog(
+                    title = it.message.toString(),
+                    description = it.cause.toString()
+                )
+            }
+        }
+
+        // Create an offline map job with the download directory path and parameters and start the job
+        offlineMapJob = offlineMapTask.createGenerateOfflineMapJob(
+            parameters = generateOfflineMapParameters,
+            downloadDirectoryPath = offlineMapPath
+        )
+
+        runOfflineMapJob()
+
+    }
+
+    /**
+     * Starts the [offlineMapJob], shows the progress dialog and
+     * displays the result offline map to the MapView
+     */
+    private fun runOfflineMapJob() {
+
+        offlineMapJob?.let { offlineMapJob ->
+
+            // Show the Job Progress Dialog
+            showJobProgressDialog.value = true
+
+            // Create a flow-collection for the job's progress
+            viewModelScope.launch {
```

Comment:
Not a required change, but when it comes to launching multiple coroutine scopes the `with` property could be helpful:

```kotlin
with(viewModelScope) {
    launch {
        // Collect Progress
    }
    launch {
        // Wait for Job result
    }
}
```